lib/models/package-info-cache.js

```diff
@@ -0,0 +1,1049 @@
+'use strict';
+
+/*
+ * Performance cache for information about packages (projects/addons/"apps"/modules)
+ * under an initial root directory and resolving addon/dependency links to other packages.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+
+let realFilePathCache;
+let realDirectoryPathCache;
+
+/**
+ * Resolve the real path for a file, return null if does not
+ * exist or is not a file or FIFO, return the real path otherwise.
+ *
+ * @private
+ * @method getRealFilePath
+ * @param  {String} filePath the path to the file of interest
+ * @return {String} real path or null
+ */
+function getRealFilePath(filePath) {
+  let realPath;
+
+  try {
+    realPath = realFilePathCache[filePath];
+
+    if (realPath) {
+      return realPath;
+    }
+
+    let stat = fs.statSync(filePath);
+
+    if (stat.isFile() || stat.isFIFO()) {
+      realPath = fs.realpathSync(filePath);
+    }
+  } catch (e) {
+    if (
+      e !== null &&
+      typeof e === 'object' &&
+      (e.code === 'ENOENT' || e.code === 'ENOTDIR')
+    ) {
+      realPath = null;
+    } else {
+      throw e;
+    }
+  }
+
+  realFilePathCache[filePath] = realPath;
+  return realPath;
+}
+
+/**
+ * Resolve the real path for a directory, return null if does not
+ * exist or is not a directory, return the real path otherwise.
+ *
+ * @private
+ * @method getRealDirectoryPath
+ * @param  {String} directoryPath the path to the directory of interest
+ * @return {String} real path or null
+ */
+function getRealDirectoryPath(directoryPath) {
```

Comment:
needs unit test

Reply:
Asked in various places. ;-) - How do you test a private function?

Reply:
Exposed the function as part of module.exports, and added tests for getRealDirectoryPath and getRealFilePath.